src/OrleansConsulUtils/SerializableMembershipTypes.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Consul;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Orleans.Runtime.Host
+{
+    public class ConsulSiloRegistration
+    {
+        private static String DeploymentKVPrefix = "orleans";  //Ensures a root KV namespace in Consul
+        private static Char KeySeparator = '/';
+
+        //Internal properties are used to hold the KV.Key and CAS value
+        internal String DeploymentId { get; set; }  //Unique Key 
+        internal SiloAddress Address { get; set; }  //Unique Key
+        internal ulong LastIndex { get; set; }      //Used to compare CAS value on update
+
+        //Public properties are serialized to the KV.Value
+        public String Hostname { get; set; }
+        public Int32 ProxyPort { get; set; }
+        public DateTime StartTime { get; set; }
+        public DateTime IAmAliveTime { get; set; }
+        public SiloStatus Status { get; set; }
+        public List<SuspectingSilo> SuspectingSilos { get; set; }
+
+        public ConsulSiloRegistration()
+        {
+            SuspectingSilos = new List<SuspectingSilo>();
+        }
+
+        internal static String ParseDeploymentKVPrefix(String deploymentId)
+        {
+            return String.Format("{0}{1}{2}", DeploymentKVPrefix, KeySeparator, deploymentId);
+        }
+
+        internal static String ParseDeploymentSiloKey(String deploymentId, SiloAddress siloAddress)
+        {
+            return String.Format("{0}{1}{2}", ParseDeploymentKVPrefix(deploymentId), KeySeparator, siloAddress.ToParsableString());
+        }
+
+        internal static ConsulSiloRegistration FromKVPair(String deploymentId, KVPair kv)
+        {
+            var ret = JsonConvert.DeserializeObject<ConsulSiloRegistration>(Encoding.UTF8.GetString(kv.Value));
```

Comment:
Its up to you, buy I usually prefer to have JSON serializer be declared explicitly. So basically adding a JsonProperty attribute on any serialized fields (or implement ISerailizable or Converter). Otherwise, Json will serialize only public properties and this is very fragile to changes - someone will change some field to private late and it will break in a very weird way, since it will just miss the field on deserialize. Happened to us in the past. 
Up to you.